### PR TITLE
3732 Update TTA module based on latest features

### LINKED ISSUE #3732: Test time augmentation need to update with new features
**Is your feature request related to a problem? Please describe.**
Hi @rijobro ,

Thanks for providing the TTA module since MONAI 0.5.
@dongyang0122 @holgerroth and I discussed the TTA module for evaluation cases, and found several things should be updated:
1. All the invert transform logic in TTA can be replaced by `Invertd` transform directly now.
And `Invertd` transform provides more features.
2. The post transform example in the doc-string is not `decollate` logic.
3. Better to ensure `determinism` for the random transforms in the example of doc-string.
4. It copy between Numpy and Tensor several times because we didn't support Tensor in transforms and don't have the proper numpy <-> tensor conversion before:
https://github.com/Project-MONAI/MONAI/blob/dev/monai/data/test_time_augmentation.py#L188
https://github.com/Project-MONAI/MONAI/blob/dev/monai/data/test_time_augmentation.py#L190
It will also cause numpy error if the transforms inverse() outputs Tensor.
Should be updated to support `NdarrayOrTensor` directly.
5. Some doc-strings exceed 120 chars.

### PR DESCRIPTION
**Pull request by Nic-Ma**
Fixes #3732 .

### Description
This PR updated the TTA module with newer MONAI features based on the discussion with @dongyang0122 and @holgerroth .

### Status
**Ready**

### Types of changes

- [x] Non-breaking change (fix or new feature that would not break existing functionality).
- [ ] Breaking change (fix or new feature that would cause existing functionality to change).
- [ ] New tests added to cover the changes.
- [ ] Integration tests passed locally by running `./runtests.sh -f -u --net --coverage`.
- [ ] Quick tests passed locally by running `./runtests.sh --quick --unittests  --disttests`.
- [ ] In-line docstrings updated.
- [ ] Documentation updated, tested `make html` command in the `docs/` folder.

### CODE REVIEW

File: monai/data/test_time_augmentation.py [hunk 1]
```diff
@@ -171,56 +186,23 @@ def __call__(
         ds = Dataset(data_in, self.transform)
         dl = DataLoader(ds, num_workers=self.num_workers, batch_size=self.batch_size, collate_fn=pad_list_data_collate)
 
-        transform_key = InvertibleTransform.trace_key(self.orig_key)
-
-        # create inverter
-        inverter = BatchInverseTransform(self.transform, dl, collate_fn=list_data_collate)
```
Comment (wyli) [inline monai/data/test_time_augmentation.py:177]: so the `BatchInverseTransform` is not used here, anymore is it deprecating?

Comment (Nic-Ma) [reply monai/data/test_time_augmentation.py:177]: Hi @wyli ,

Thanks for your review.
We already dropped `BatchInverseTransform` in `Invertd` transform and `TransformInverter` handler, and for most cases `num_workers=0` is fastest in `BatchInverseTransform`, but I think it provides the ability to execute extremely huge inverse operations in multi-processing:
https://github.com/Project-MONAI/MONAI/blob/dev/monai/transforms/inverse_batch_transform.py#L96
So maybe let's keep it for now?
@rijobro , what do you think? Do you have any other special use cases?

Thanks in advance.

Comment (rijobro) [reply monai/data/test_time_augmentation.py:177]: I can't think of any other use cases, so perhaps we should keep `BatchInverseTransform`, but use `Invertd` in the vast majority of other cases.

Comment (Nic-Ma) [reply monai/data/test_time_augmentation.py:177]: @rijobro thanks for your review!

File: monai/data/test_time_augmentation.py [hunk 2]
```diff
@@ -109,6 +112,9 @@ def __init__(
         nearest_interp: bool = True,
         orig_meta_keys: Optional[str] = None,
         meta_key_postfix=DEFAULT_POST_FIX,
+        to_tensor: bool = True,
+        output_device: Union[str, torch.device] = "cpu",
+        post_func: Callable = lambda x: x,
```
Comment (rijobro) [inline monai/data/test_time_augmentation.py:117]:
```suggestion
        post_func: Callable = _identity,
```

Comment (Nic-Ma) [reply monai/data/test_time_augmentation.py]: Hi @rijobro ,

Sure, updated in the latest commit.

Thanks.

**Comment by rijobro** [comment]
Hi @Nic-Ma, I've created a PR to add to your PR.

1. Improve tests
2. Reduce data type conversion.

The test currently fails, I will try to debug on Monday, but I feel it is important that it works before we merge. We seem to get different results for:

1. numpy and torch inputs, and
2. using the `post_func` instead of `inferrer_fn=post_trans(model(x))`.

**Comment by Nic-Ma** [comment]
Hi @rijobro ,

Plan sounds good to me!

Thanks for your help here.

**Review by wyli — APPROVED** [review]
Thanks, it looks good to me.

**Comment by rijobro** [comment]
@Nic-Ma my PR is ready: Nic-Ma#362.

Think you will need to merge it for the unit tests to work.

**Comment by Nic-Ma** [comment]
> @Nic-Ma my PR is ready: [Nic-Ma#362](https://github.com/Nic-Ma/MONAI/pull/362).
> 
> Think you will need to merge it for the unit tests to work.

Hi @rijobro ,

Thanks for your enhancement, started CI tests in your PR and put some minor comments in it.

**Review by wyli — APPROVED** [review]
Thanks, it looks good to me. Minor issues (not introduced by this PR) are 1. Runtime error could be a warning to allow for randomness defined in the network such as dropout. 2 np.std has a ddof parameters but this is currently hardcoded cc @rijobro

**Comment by Nic-Ma** [comment]
Hi @wyli ,

Thanks for your review.
1. I changed `RuntimeError` to `Warning` and updated the unit tests in the latest commit.
2. About the `std` arg, PyTorch also has another arg `unbiased`:
https://pytorch.org/docs/stable/generated/torch.std.html
I think maybe we don't need to add these additional args so far? If we got some requests, let's add it then?

What do you think?

Thanks.
